SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -412,6 +413,37 @@ bool SaveSchemaDatabase(const FString& PackagePath)
 	SchemaDatabase->ComponentIdToClassPath = CreateComponentIdToClassPathMap();
 	SchemaDatabase->LevelComponentIds = LevelComponentIds;
 
+	FString CompiledSchemaDir = FPaths::Combine(SpatialGDKServicesConstants::SpatialOSDirectory, TEXT("build/assembly/schema"));
+
+	// Generate hash
+	{
+		SchemaDatabase->SchemaDescriptorHash = 0; // If we are in a unusual environment (test) these paths are incorrect 
+		FString DescriptorPath = FPaths::Combine(CompiledSchemaDir, TEXT("schema.descriptor"));
```

Comment:
So the `schema.descriptor` is pumped out when running the `SchemaCompiler` in the function `RunSchemaCompiler();`, this is called as the last part of `SpatialGDKGenerateSchema()`. Doesn't this mean when saving the schema database you are generating the schema hash for the old descriptor? We need to generate the descriptor and then the hash.